### PR TITLE
Initialized associations pull request

### PR DESCRIPTION
**Pull request by AleWaweru**
- [ ] created associations between Person, Classroom, Student, Book, and Rental classes.

### CODE REVIEW

File: .github/workflows/book.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    def initialize(title, author)
+      @title = title
+      @author = author
+      @rentals = []
+    end
+  
+    def add_rental(date, person)
+      rental = Rental.new(date, person, self)
+      @rentals << rental
+      person.add_rental(rental)
```
Comment (DammyShittu) [inline .github/workflows/book.rb:13]: - [ ] The Person class should have an independent `add_rental` method like the Book class for creating the association between both classes. So writing this method this way is not a good idea. Kindly remove the second and third lines in this method. We would refactor the Rental class to add itself to the `@rentals` of Book and Person, along with the Person class. For now, kindly change this method to this:

  ```
  def add_rental(date, person)
    Rental.new(date, person, self)
  end
  ```

File: .github/workflows/rental.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Rental
+    attr_accessor :date, :person, :book
+  
+    def initialize(date, person, book)
+      @date = date
+      @person = person
+      @book = book
+    end
```
Comment (DammyShittu) [inline .github/workflows/rental.rb:8]: - [ ] Let's modify the Rental class to add itself to the rentals arrays of the classes.

  ```
  def initialize(date, person, book)
    @date = date

    @person = person
    person.rentals << self

    @book = book
    book.rentals << self
  end
  ```

File: .github/workflows/person.rb [hunk 3]
```diff
@@ -1,49 +1,36 @@
...
+    @age = age
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+
+  def add_rental(rental)
+    @rentals << rental
+  end
```
Comment (DammyShittu) [inline .github/workflows/person.rb:25]: - [ ] Let's make the Person class `add_rental` to be independent of the Book class, so we can use it without having to call it from the book class.

  ```
  def add_rental(date, book)
    Rental.new(date, self, book)
  end
  ```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @AleWaweru,

## Highlights
✔️ Classroom and student association

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: .github/workflows/person.rb (hunk 3 above)
Comment (AleWaweru) [reply .github/workflows/person.rb:25]: done

File: .github/workflows/rental.rb (hunk 2 above)
Comment (AleWaweru) [reply .github/workflows/rental.rb:8]: done

File: .github/workflows/book.rb (hunk 1 above)
Comment (AleWaweru) [reply .github/workflows/book.rb:13]: done

**Review by prolajumokeoni — APPROVED** [review]
Hi @AleWaweru 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
  - All changes requested have been implemented ✅ 
 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**